### ISSUE TITLE
Invalid value of backpressure time metric observed, seems like a computation error

### ISSUE THREAD

**Issue by ashvina** [twitter/heron#1753]
In a local experiment, we deployed a topology with a few spouts and bolts (around 5 each). The bolts were able to process all the tuples and hence we did not observe any backpressure. Subsequently, we increased the number of spouts by 4 times (to 20) using the `update` command. As expected, this resulted in backpressure. We then queried the tracker for `__time_spent_back_pressure_by_compid` metric for a 60 second interval for all instances of a component.

E.g. query: `host:8888/topologies/metrics?cluster=yarn&environ=default&topology=ET&interval=60&metricname=__time_spent_back_pressure_by_compid/container_1_exclaim_1&component=__stmgr__`

When we added the backpressure time for all instances of a component, we observed that the total time spent in backpressure was around `250k` (assuming the unit is milliseconds). This seems invalid as time spent in back pressure in a 60 second interval should not be more than 60k milliseconds (see graphite chart below)

![img](https://cloud.githubusercontent.com/assets/10034060/23683408/b8ee8b8e-034d-11e7-8ed1-a1d18a32d229.png)

**Comment by congwang** [comment]
@ashvina I must miss something here... When you add them up, the max should be N*60k in a 60 second interval, where N is the number of instances you add. No?

**Comment by avflor** [comment]
@congwang I don't think it should be more than 60 sec since only one instance of a given component would be the source of backpressure at a time. So during a 1-minute interval, the sum should not be more than 60 sec. @maosongfu may want to comment.

**Comment by maosongfu** [comment]
I think this metric should not exceed 60s. This value is directly reported by stmgr so there might be some calculation errors in stmgr.

**Comment by huijunw** [comment]
At present, the metric timestamp works as follow. The source (eg. spout/bolt/stmgr) sends the metric with no timestamp to the metricsMgr. The metricMgr creates metric msg with local time and send to tmaster-collector where the timestamp is dropped. Tmaster-collector marks the metric with tmaster local time.
Ideally, IMO, the metric timestamp is supposed to be tagged by the source where the metric is generated.

One possible cause to the over 60k backpressure time is: several backpressure metric msgs belonging to several minutes may arrive at the tmaster-colector in one 60s so that they count to more than 60k.
